### PR TITLE
Add 'issue' and 'reason' parameters to FailingTest.

### PR DESCRIPTION
**Pull request by scheglov**

### CODE REVIEW

File: lib/test_reflective_loader.dart [hunk 1]
```diff
@@ -269,8 +269,11 @@ typedef dynamic _TestFunction();
 class FailingTest {
   /**
    * Initialize this annotation with the given issue URI.
```
Comment (devoncarew) [inline lib/test_reflective_loader.dart]: `with the given issue URI` ==> `with the given arguments`

File: lib/test_reflective_loader.dart [hunk 2]
```diff
@@ -269,8 +269,11 @@ typedef dynamic _TestFunction();
...
+   *
+   * [issue] is a full URI describing the failure and used for tracking.
```
Comment (devoncarew) [inline lib/test_reflective_loader.dart]: reason first, and then issue?

Comment (bwilkerson) [reply lib/test_reflective_loader.dart]: Alphabetic order lgtm.

Comment (bwilkerson) [reply lib/test_reflective_loader.dart]: If there is an unnamed parameter, and you only want to include an issue number, then you have to write:
```
@FailingTest('', issue: '...')
```
Which seems silly. I suspect that this is one of the reasons Flutter chose to make all parameters named, even when some are required,

**Pull request merged**